### PR TITLE
Id 68 post incident

### PR DESCRIPTION
**Pull request by wojtek717**

### CODE REVIEW

File: kpz-mpk/APIServices/ApiService.swift [hunk 1]
```diff
@@ -26,6 +26,7 @@ class ApiService: ApiServiceProtocol {
     var request: URLRequest = URLRequest(url: url.appendingPathComponent(endpoint.path))
     
     request.httpMethod = endpoint.method.rawValue
+    request.allHTTPHeaderFields = ["Authorization": "Barer \(SessionManager.shared.credentials?.accessToken)"]
```
Comment (finalsrebrny) [inline kpz-mpk/APIServices/ApiService.swift]: Zmienne typu "Authorization" powinny znajdować się w Constants

File: kpz-mpk/APIServices/IncidentApiService.swift [hunk 2]
```diff
@@ -11,6 +11,7 @@ import Foundation
 protocol IncidentApiServiceProtocole {
   func getIncidents(success: (([Incident]) -> ())?, failure: ((ApiError) -> Void)?)
   func getIncident(id: String, success: ((Incident) -> ())?, failure: ((ApiError) -> Void)?)
+  func psotIncident(incidentToReport incident: ReportIncident, success: ((Incident) -> ())?, failure: ((ApiError) -> Void)?)
```
Comment (finalsrebrny) [inline kpz-mpk/APIServices/IncidentApiService.swift]: psotIncident => postIncident

File: kpz-mpk/APIServices/IncidentApiService.swift [hunk 3]
```diff
@@ -23,4 +24,8 @@ class IncidentApiService: IncidentApiServiceProtocole {
   func getIncidents(success: (([Incident]) -> ())?, failure: ((ApiError) -> Void)?) {
     apiService.request(endpoint: IncidentEndpoint.getIncidents, success: success, failure: failure)
   }
+  
+  func psotIncident(incidentToReport incident: ReportIncident, success: ((Incident) -> ())?, failure: ((ApiError) -> Void)?) {
```
Comment (finalsrebrny) [inline kpz-mpk/APIServices/IncidentApiService.swift]: psotIncident => postIncident

Comment (wojtek717) [reply kpz-mpk/APIServices/IncidentApiService.swift]: This is outdated

File: kpz-mpk/APIServices/IncidentApiService.swift (hunk 2 above)
Comment (wojtek717) [reply kpz-mpk/APIServices/IncidentApiService.swift]: This is outdated

File: kpz-mpk/kpz-mpk/LinePickView/ViewControllers/RoutePickViewController.swift [hunk 4]
```diff
@@ -20,9 +21,22 @@ class RoutePickViewController: UITableViewController {
...
+        return
+    }
+    
+    viewModel.reportIncident(routeId: routeId, tripHeadsign: tripHeadsign)
+  }
 }
 
 extension RoutePickViewController: RoutePickViewControllerPresenter {
+  func dismissView() {
+    self.view.window?.rootViewController?.dismiss(animated: true, completion: nil)
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/LinePickView/ViewControllers/RoutePickViewController.swift:38]: It's not a good practice to use `UIWindow` instance without any reason. If you tried to dismiss all presented VC's, you should call `.dismiss` on a Map VC

File: kpz-mpk/kpz-mpk/LinePickView/ViewModels/RoutePickViewModel.swift [hunk 5]
```diff
@@ -12,20 +12,24 @@ import UIKit
...
 
 final class RoutePickViewModel {
+  weak var mapViewDelegate: MapViewDelegate?
   private var reportedLocation: CLLocationCoordinate2D
   private var reportedType: IncidenType
   private weak var presenter: RoutePickViewControllerPresenter?
   
   let routeApiService: RouteApiServiceProtocole = RouteApiService()
-
+  let incidentApiService: IncidentApiService = IncidentApiService()
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/LinePickView/ViewModels/RoutePickViewModel.swift:21]: private

File: kpz-mpk/kpz-mpk/Map/Delegates/MapViewDelegate.swift [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+//  kpz-mpk
+//
+//  Created by Wojciech Konury on 13/05/2020.
+//  Copyright © 2020 kpz-mpk. All rights reserved.
+//
+
+import Foundation
+
+protocol MapViewDelegate: class {
+  func mapView(incidentIsReported incident: Incident)
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/Map/Delegates/MapViewDelegate.swift:12]: Delegate tells about reporting a new incident. Maybe `IncidentDelegate` is a better name?

File: kpz-mpk/APIServices/ApiService.swift [hunk 7]
```diff
@@ -15,6 +15,12 @@ protocol ApiServiceProtocol {
 
 class ApiService: ApiServiceProtocol {
   
+  let sessionManager: SessionManager
```
Comment (leszko11) [inline kpz-mpk/APIServices/ApiService.swift:18]: private

File: kpz-mpk/kpz-mpk/Map/Delegates/IncidentDelegate.swift [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+//  kpz-mpk
+//
+//  Created by Wojciech Konury on 13/05/2020.
+//  Copyright © 2020 kpz-mpk. All rights reserved.
+//
+
+import Foundation
+
+protocol IncidentDelegate: class {
+  func incidentDelegate(incidentIsReported incident: Incident)
```
Comment (leszko11) [inline kpz-mpk/kpz-mpk/Map/Delegates/IncidentDelegate.swift:12]: Still we can improve naming here
https://github.com/raywenderlich/swift-style-guide#delegates

**Pull request merged**